src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -59,6 +66,43 @@ public function process(ContainerBuilder $container)
                 $tmpContainer->addObjectResource($configuration);
             }
 
+            if (null !== $envTypes && method_exists(BaseNode::class, 'setPlaceholder')) {
+                $defaultEnvBag = new ParameterBag($parameters);
+                foreach ($resolvingBag->getEnvPlaceholders() as $env => $placeholders) {
+                    $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+                    $types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string');
+                    $values = array();
+                    $default = null;
+                    if ($hasEnv = (false === $i && $defaultEnvBag->has("env($env)"))) {
+                        switch ($type = gettype($default = $defaultEnvBag->get("env($env)"))) {
+                            case 'boolean': $type = 'bool'; break;
```

Comment:
should we rename our allowedTypes to match those of gettype and remove that switch?

Reply:
Think so :+1: preferring FQ over short names.

Reply:
While at it should we infer `boolean|integer|float|string` from `scalar` while resolving? Nice to have perhaps.